src/ConversationList/ConversationListItem.vala

```diff
@@ -32,12 +32,36 @@ public class Mail.ConversationListItem : Gtk.ListBoxRow {
 
     public int64 timestamp {
         get {
+            int64 timestamp = 0;
             if (node.message.date_received == 0) {
                 // Sent messages do not have a date_received timestamp.
                 return node.message.date_sent;
+                timestamp = node.message.date_sent;
```

Comment:
This will not be reached because the `return` statement on the previous line is still there.